pallets/fusion/src/lib.rs

```diff
@@ -0,0 +1,1989 @@
+			Self::deposit_event(Event::PoolDeleted { pool_id });
+		}
+
+		Ok(())
+	}
+
+	/// Setup the fusion currency rates for the new era
+	fn setup_currency_rates(era: EraIndex) -> DispatchResult {
+		for (currency_id, currency) in FusionCurrencies::<T>::iter() {
+			// Skip if the currency is destroyed
+			if currency.is_destroyed {
+				continue;
+			}
+
+			// Try to get the new rate from the rate changes storage
+			let new_rate = FusionCurrencyRateChanges::<T>::get(currency_id).or_else(|| {
+				// Fallback to the current era's rate
+				FusionCurrencyRates::<T>::get(currency_id, era)
+			});
+
+			// If neither a new rate nor a current rate is found, trigger an error
+			let rate = new_rate.ok_or(Error::<T>::CurrencyRateNotFound)?;
+
+			// Insert the rate for the next era
+			FusionCurrencyRates::<T>::insert(currency_id, era + 1, rate);
+		}
+		Ok(())
+	}
+
+	/// Clean history depth storages and send old pending rewards to 'RewardRemainder'
+	fn clean_history_depth_storages(era: EraIndex) -> DispatchResult {
+		let history_depth = T::HistoryDepth::get();
+
+		if let Some(era_to_clear) = era.checked_sub(history_depth) {
+			// Clean fusion exposures
+			let pool_count = FusionPoolCount::<T>::get();
+			let _ = FusionExposures::<T>::clear_prefix(era_to_clear, pool_count, None);
+
+			// Clean currency rates
+			FusionCurrencyRates::<T>::iter_keys().for_each(|(currency_id, era)| {
+				if era == era_to_clear {
+					FusionCurrencyRates::<T>::remove(currency_id, era);
+				}
+			});
+
+			// Clean claimed rewards
+			ClaimedRewards::<T>::iter_keys().for_each(|(evm_address, pool_id, era)| {
+				if era == era_to_clear {
+					ClaimedRewards::<T>::remove((evm_address, pool_id, era));
+				}
+			});
+
+			// Clean fusion era rewards and compute remaining rewards
+			let mut remaining_rewards = BalanceOf::<T>::zero();
+			FusionEraRewards::<T>::drain_prefix(era).for_each(|(_, rewards)| {
+				remaining_rewards = remaining_rewards
+					.saturating_add(rewards.rewards.saturating_sub(rewards.claimed_rewards));
+			});
+
+			// Send rewards that were not claimed from claimable account to RewardRemainder
+			if remaining_rewards > 0u32.into() {
+				let claimable_reward_account = Self::accounts().claimable_reward_account;
+				let imbalance = T::Currency::withdraw(
+					&claimable_reward_account,
+					remaining_rewards,
+					WithdrawReasons::all(),
+					ExistenceRequirement::AllowDeath,
+				)?;
+				T::RewardRemainder::on_unbalanced(imbalance);
+				Self::deposit_event(Event::RewardRemainderSent {
+					amount: remaining_rewards,
+				});
+			}
+		}
+
+		Ok(())
+	}
+
+	/// Compute rewards for each pool and set them in storage
+	/// Exposure was set at the end of the era N for era N
+	/// Reward computatation is done at the end of era N for era N-1
+	fn compute_era_rewards(era: EraIndex, era_duration: u64) -> DispatchResult {
+		if let Some(era_to_process) = era.checked_sub(1) {
```

Comment:
Too many indentations makes it hard to reason with it.
For this line you can do the following
```rust
let Some(era_to_process) = era.checked_sub(1) else {
    return Ok(());
};
```